docs/reference/index-modules/indexing-pressure.asciidoc

```diff
@@ -0,0 +1,56 @@
+[[index-modules-indexing-pressure]]
+== Indexing Pressure
+
+Indexing documents into Elasticsearch introduces system load in the form of
+memory and CPU load. Each indexing operation includes coordinating, primary, and
+replica stages. These stages can be performed across multiple nodes in the
+cluster. If too much indexing work is introduced into the system, the cluster
+can become saturated.
+
+Indexing pressure is primarily generated by external operations such as indexing
+requests or CCR follow tasks. However, there is some internal indexing pressure
+during a shard recovery or primary failover.
+
+Elasticsearch internally monitors indexing load. When the load exceeds
+certain limits, new indexing work will be rejected.
+
+[float]
+=== Indexing Stages
```

Comment:
The stages are sequential (first coordination, then primary, then replica).

What's missing in this doc is how the accounting works, i.e. that coordination accounting is only done once all subsequent stages are done, same for primary stage. This can be explained that ES internally retries operations and therefore keeps the operation in-memory until all subsequent stages have successfully completed.